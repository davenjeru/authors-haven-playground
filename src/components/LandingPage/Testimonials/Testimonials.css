img.testimonial {
    width: 192px !important;
    height: 192px !important;
}

.home-banner{
    background: url('../../../assets/images/books.jpg') space;
}
