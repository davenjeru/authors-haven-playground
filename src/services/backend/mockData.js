// initialize the user information that should be sent in the request body in the sign-up route
export const mockUserInfo = {
  email: 'test@email.com',
  username: 'testUsername',
  password: 'testPassword',
};

// model the response object according to our backend specification
// notice the capital M in Message
export const mockSuccessResponseObject = {
  data:
    { Message: 'Check your email for further instructions' },
};

// the backend models the errors into an object whose attributes are arrays
// the front end should extract the first error
export const mockErrorResponseObject = {
  data: {
    errors: {
      email: ['Email already exists, please login or use a different email'],
      username: ['Username already exists, please enter a different username'],
    },
  },
};
