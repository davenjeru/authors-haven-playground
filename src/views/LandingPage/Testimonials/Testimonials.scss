img {
  &.testimonial {
    width: 60%;
  }
}
