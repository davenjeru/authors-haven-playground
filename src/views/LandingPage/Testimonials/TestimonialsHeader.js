import React from 'react';

const TestimonialsHeader = () => (
  <div className="row">
    <div className="col-md-12">
      <h1 className="text-primary">Testimonials</h1>
      <p className="lead">We grow together with the community</p>
    </div>
  </div>
);

export default TestimonialsHeader;
