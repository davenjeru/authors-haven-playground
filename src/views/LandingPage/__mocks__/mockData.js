import Chimammanda from '../../../assets/images/face1.jpg';
import Hos from '../../../assets/images/hos.jpg';
import Ruth from '../../../assets/images/ruth.jpg';

export const benefitsArray = [
  {
    value: 'Authentic', emoji: 'ðŸ‘Œ',
  },
  {
    value: 'Knowledgeable Authors', emoji: 'ðŸ“š',
  },
  {
    value: 'Membership', emoji: 'ðŸ’³',
  },
  {
    value: 'Affordable', emoji: 'ðŸ’¸',
  },
  {
    value: 'Amazing', emoji: 'ðŸ˜Œ',
  },
  {
    value: 'Fun', emoji: 'ðŸ™Œ',
  },
];

export const featuredCategories = [
  {
    name: 'Technology',
    count: 312,
  },
  {
    name: 'Lifestyle',
    count: 299,
  },
  {
    name: 'Science',
    count: 259,
  },
  {
    name: 'Finance',
    count: 219,
  },
  {
    name: 'Politics',
    count: 155,
  },
];

export const testimonialsArray = [
  {
    name: 'Chimammanda',
    text: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa.',
    image: Chimammanda,
    role: 'Senior writer',
  },
  {
    name: 'Hoslack',
    text: 'Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec.',
    image: Hos,
    role: 'Reader',
  },
  {
    name: 'Ruth',
    text: 'Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Cum sociis natoque penatibus et magnis dis parturient montes.',
    image: Ruth,
    role: 'Reader',
  },
];

const mockData = {
  benefitsArray,
  featuredCategories,
  testimonialsArray,
};

export default mockData;
