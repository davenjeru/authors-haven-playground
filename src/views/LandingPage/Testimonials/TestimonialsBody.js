import React from 'react';
import PropTypes from 'prop-types';
import './Testimonials.css';

const TestimonialsBody = ({ testimonialsArray }) => (
  <div className="row">
    {testimonialsArray.map(testimonial => (
      <div key={testimonial.name} className="col-md-4 p-4">
        <img
          className="img-fluid rounded-circle testimonial"
          src={testimonial.image.default || testimonial.image}
          alt={testimonial.name}
        />
        <p className="my-4">
          <i>
            {testimonial.text}
          </i>
        </p>
        <p>
          <b>{testimonial.name}</b>
          <br />
          {testimonial.role}
        </p>
      </div>
    ))}
  </div>
);

TestimonialsBody.propTypes = {
  testimonialsArray: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      text: PropTypes.string,
      image: PropTypes.string,
      role: PropTypes.string,

    }),
  ).isRequired,
};

export default TestimonialsBody;
