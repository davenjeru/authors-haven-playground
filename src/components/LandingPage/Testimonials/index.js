import React from 'react';
import TestimonialsBody from './TestimonialsBody';
import TestimonialsHeader from './TestimonialsHeader';

const Testimonials = ({ testimonialsArray }) => (
  <div className="py-5 text-center bg-light">
    <div className="container">
      <TestimonialsHeader />
      <TestimonialsBody testimonialsArray={testimonialsArray} />
    </div>
  </div>
);

Testimonials.propTypes = {
  testimonialsArray: TestimonialsBody.propTypes.testimonialsArray.isRequired,
};

export default Testimonials;
