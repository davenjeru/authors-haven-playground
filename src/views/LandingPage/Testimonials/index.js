import React from 'react';
import TestimonialsBody from './TestimonialsBody';
import TestimonialsHeader from './TestimonialsHeader';

const Testimonials = ({ testimonialsArray }) => (
  <div className="py-5 text-center bg-light">
    <div className="container">
      <TestimonialsHeader />
      <TestimonialsBody testimonialsArray={testimonialsArray} />
    </div>
  </div>
);

Testimonials.propTypes = {
  ...TestimonialsBody.propTypes,
};

export default Testimonials;
