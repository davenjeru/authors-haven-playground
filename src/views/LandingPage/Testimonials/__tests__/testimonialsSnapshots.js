import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import React from 'react';
import Testimonials from '..';
import { testimonialsArray } from '../../__mocks__/mockData';

describe('The Testimonials', () => {
  it('should not regress', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Testimonials testimonialsArray={testimonialsArray} />
      </MemoryRouter>);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
