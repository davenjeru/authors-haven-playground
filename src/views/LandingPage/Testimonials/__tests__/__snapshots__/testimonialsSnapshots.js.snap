// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Testimonials should not regress 1`] = `
<div
  className="py-5 text-center bg-light"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-12"
      >
        <h1
          className="text-primary"
        >
          Testimonials
        </h1>
        <p
          className="lead"
        >
          We grow together with the community
        </p>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-md-4 p-4"
      >
        <img
          alt="Chimammanda"
          className="img-fluid rounded-circle testimonial"
          src="face1.jpg"
        />
        <p
          className="my-4"
        >
          <i>
            Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa.
          </i>
        </p>
        <p>
          <b>
            Chimammanda
          </b>
          <br />
          Senior writer
        </p>
      </div>
      <div
        className="col-md-4 p-4"
      >
        <img
          alt="Hoslack"
          className="img-fluid rounded-circle testimonial"
          src="hos.jpg"
        />
        <p
          className="my-4"
        >
          <i>
            Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec.
          </i>
        </p>
        <p>
          <b>
            Hoslack
          </b>
          <br />
          Reader
        </p>
      </div>
      <div
        className="col-md-4 p-4"
      >
        <img
          alt="Ruth"
          className="img-fluid rounded-circle testimonial"
          src="ruth.jpg"
        />
        <p
          className="my-4"
        >
          <i>
            Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Cum sociis natoque penatibus et magnis dis parturient montes.
          </i>
        </p>
        <p>
          <b>
            Ruth
          </b>
          <br />
          Reader
        </p>
      </div>
    </div>
  </div>
</div>
`;
